node/src/blockchain/blockchain_interface/blockchain_interface_web3/lower_level_interface_web3.rs

```diff
@@ -1,105 +1,33 @@
-}
-
-impl FromStr for TxStatus {
-    type Err = String;
-
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
-        match s {
-            "Pending" => Ok(TxStatus::Pending),
-            "Failed" => Ok(TxStatus::Failed), // TODO: GH-631: This should be removed
-            s if s.starts_with("Succeeded") => {
-                // The format is "Succeeded(block_number, block_hash)"
-                let parts: Vec<&str> = s[10..s.len() - 1].split(',').collect();
-                if parts.len() != 2 {
-                    return Err("Invalid Succeeded format".to_string());
-                }
-                let block_number: u64 = parts[0]
-                    .parse()
-                    .map_err(|_| "Invalid block number".to_string())?;
-                let block_hash =
-                    H256::from_str(&parts[1][2..]).map_err(|_| "Invalid block hash".to_string())?;
-                Ok(TxStatus::Succeeded(TransactionBlock {
-                    block_hash,
-                    block_number: U64::from(block_number),
-                }))
-            }
-            _ => Err(format!("Unknown status: {}", s)),
-        }
-    }
-}
-
-impl Display for TxStatus {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        match self {
-            TxStatus::Failed => write!(f, "Failed"),
-            TxStatus::Pending => write!(f, "Pending"),
-            TxStatus::Succeeded(block) => {
-                write!(
-                    f,
-                    "Succeeded({},{:?})",
-                    block.block_number, block.block_hash
-                )
-            }
-        }
-    }
-}
-
-#[derive(Debug, PartialEq, Eq, Clone)]
-pub struct TxReceipt {
-    pub transaction_hash: H256,
-    pub status: TxStatus,
-}
-
-#[derive(Debug, Default, PartialEq, Eq, Clone, Copy, Serialize, Deserialize)]
-pub struct TransactionBlock {
-    pub block_hash: H256,
-    pub block_number: U64,
-}
-
-impl From<TransactionReceipt> for TxReceipt {
+impl From<TransactionReceipt> for StatusReadFromReceiptCheck {
     fn from(receipt: TransactionReceipt) -> Self {
-        let status = match (receipt.status, receipt.block_hash, receipt.block_number) {
+        match (receipt.status, receipt.block_hash, receipt.block_number) {
             (Some(status), Some(block_hash), Some(block_number)) if status == U64::from(1) => {
-                TxStatus::Succeeded(TransactionBlock {
+                StatusReadFromReceiptCheck::Succeeded(TxBlock {
                     block_hash,
                     block_number,
                 })
             }
-            (Some(status), _, _) if status == U64::from(0) => TxStatus::Failed,
-            _ => TxStatus::Pending,
-        };
-
-        TxReceipt {
-            transaction_hash: receipt.transaction_hash,
-            status,
+            (Some(status), _, _) if status == U64::from(0) => {
+                StatusReadFromReceiptCheck::Failed(BlockchainTxFailure::Unrecognized)
```

Comment:
You shouldn't be holding the `BlockchainTxFailure::Unrecognized` here inside Failed. Instead you should just return this here - `StatusReadFromReceiptCheck::Reverted`.